From 78f1d2b8d9dcf68885a6d040cd6f56c6112c549d Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Tue, 29 Jun 2021 10:00:09 +0800
Subject: [PATCH] Check the value of EpNum before accessing the EpHandles

EpNum shouldn't larger than (16 * 2), since basing on the usb
protocol, a USB device can only have up to 32 endpoints
(16 OUT and 16 IN).

Tracked-On: OAM-97425
Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 libefiusb/device_mode/XdciDWC.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libefiusb/device_mode/XdciDWC.c b/libefiusb/device_mode/XdciDWC.c
index cac538a..86d0041 100644
--- a/libefiusb/device_mode/XdciDWC.c
+++ b/libefiusb/device_mode/XdciDWC.c
@@ -2824,6 +2824,11 @@ DwcXdciEpClearStall (
   //
   EpNum = DwcXdciGetPhysicalEpNum (EpInfo->EpNum, EpInfo->EpDir);
 
+  if (EpNum >= DWC_XDCI_MAX_ENDPOINTS * 2) {
+    DEBUG ((DEBUG_INFO, "DwcXdciEpClearStall: INVALID EpNum\n"));
+    return EFI_DEVICE_ERROR;
+  }
+
   //
   // Set Ep State Info
   //
-- 
2.25.1

